linux-shell/handler.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"os/user"
+)
+
+
+func executeCMD(input string, history map[int]string, validCMD map[string]string) bool {
+
+	// split the command and arguments
+	fields := strings.Fields(input)
+	command := fields[0]
+
+	switch command {
+	case "ls":
+		executeLS()
+	case "pwd":
+		executePWD()
+	case "history":
+		executeHISTORY(history)
+	case "cd":
+		executeCD(fields)
+	case "help":
+		executeHELP(validCMD)
+	case "exit":
+		os.Exit(0)
+	default:
+		fmt.Printf("command not found: '%s'\n", fields[0])
+		fmt.Printf("type 'help' to see all available commands\n")
+	}
+
+	return true
+}
+
+// to handle 'pwd' command
+func executePWD() {
+	pwd, err := os.Getwd()
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Print errors in proper format and return from there immediately. 

Here we don't want to raise panic because we want this program to continue and user will provide another command. But we should notify the user properly that error has been occurred by printing error in error formats.

The another way is to return the error to the calling function and the calling function will handle the error based on its severity. 

The error which are of highest severity for them we need to raise panic that program could not run if that error comes.

Reply:
Same applies in all function

Reply:
Resolved